axlearn/common/array_serialization_test.py

```diff
@@ -335,71 +345,87 @@ def _verify_shard_info(
         self.assertTrue(np.all(out_array == np.array(single_device_arr)))
 
     @parameterized.product(
-        max_data_shard_degree=[1, -1, 2, 4, 8], shard_threshold_bytes=[1000 * 1000 * 1000, 1]
+        max_data_shard_degree=[1, -1, 2, 4, 8, 32, 64],
+        shard_threshold_bytes=[1000 * 1000 * 1000, 1],
     )
     @pytest.mark.skipif(
-        jax.device_count() != 8 or jax.process_count() != 1,
-        reason="Incorrect device count for mesh.",
+        jax.device_count() != (64 if jax.default_backend() == "neuron" else 8)
+        or jax.process_count() != 1,
+        reason=f"Incorrect device count ({jax.device_count()}) for the test.",
     )
     def test_shard_info_partially_replicated(
         self, max_data_shard_degree: int, shard_threshold_bytes: int
     ):
+        skip_if_mesh_too_big(max_data_shard_degree)
         single_device_arr = jnp.arange(0, 1024 * 1024).reshape(1024, 1024)
-        devices = mesh_utils.create_device_mesh((8,))
+        devices = mesh_utils.create_device_mesh((jax.device_count(),))
         sharding = PositionalSharding(devices)
 
-        arr = jax.device_put(single_device_arr, sharding.reshape(4, 2).replicate(0))
+        arr = jax.device_put(
+            single_device_arr, sharding.reshape(int(jax.device_count() / 2), 2).replicate(0)
+        )
 
         replica_count = _num_replicas_per_shard(arr)
-        self.assertEqual(replica_count[((None, None, None), (0, 512, None))], 4)
-        self.assertEqual(replica_count[((None, None, None), (512, 1024, None))], 4)
+        self.assertEqual(
+            replica_count[((None, None, None), (0, 512, None))], int(jax.device_count() / 2)
+        )
+        self.assertEqual(
+            replica_count[((None, None, None), (512, 1024, None))], int(jax.device_count() / 2)
+        )
 
         self._verify_shard_info(
             single_device_arr, arr, max_data_shard_degree, shard_threshold_bytes
         )
 
     @parameterized.product(
-        max_data_shard_degree=[1, -1, 2, 4, 8], shard_threshold_bytes=[1000 * 1000 * 1000, 1]
+        max_data_shard_degree=[1, -1, 2, 4, 8, 32, 64],
+        shard_threshold_bytes=[1000 * 1000 * 1000, 1],
     )
     @pytest.mark.skipif(
-        jax.device_count() != 8 or jax.process_count() != 1,
-        reason="Incorrect device count for mesh.",
+        jax.device_count() != (64 if jax.default_backend() == "neuron" else 8)
+        or jax.process_count() != 1,
+        reason=f"Incorrect device count ({jax.device_count()}) for the test.",
```

Comment:
New message replaces word "mesh" with "the test" - I'd keep word "mesh" in the message
e.g.
```
reason=f"Invalid device count ({jax.device_count()}) for the test mesh configuration."
```